python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,1089 @@
+
+        fluxVal = fluxVar = fitParams['flux']
+        fluxErr = fluxErrNeg = fitResult.params['flux'].stderr
+        if self.posImage is not None:
+            fluxVar = computeSumVariance(self.posImage, source.getFootprint())
+        else:
+            fluxVar = computeSumVariance(self.diffim, source.getFootprint())
+
+        fluxValNeg, fluxVarNeg = fluxVal, fluxVar
+        if separateNegParams:
+            fluxValNeg = fitParams['fluxNeg']
+            fluxErrNeg = fitResult.params['fluxNeg'].stderr
+        if self.negImage is not None:
+            fluxVarNeg = computeSumVariance(self.negImage, source.getFootprint())
+
+        try:
+            signalToNoise = np.sqrt((fluxVal/fluxVar)**2 + (fluxValNeg/fluxVarNeg)**2)
+        except:  # catch divide by zero - should never happen.
+            signalToNoise = np.nan
+
+        out = DipoleFitAlgorithm.resultsOutput(
+            fitParams['xcenPos'], fitParams['ycenPos'], fitParams['xcenNeg'], fitParams['ycenNeg'],
+            fluxVal, -fluxValNeg, fluxErr, fluxErrNeg, centroid[0], centroid[1], angle,
+            signalToNoise, fitResult.chisqr, fitResult.redchi)
+
+        if return_fitObj:  # for debugging
+            return out, fitResult
+        return out
+
+    def displayFitResults(self, result, footprint):
+        """Usage: fig = displayFitResults(result, fp)"""
+        try:
+            # Display data, model fits and residuals (currently uses matplotlib display functions)
+            import matplotlib.pyplot as plt
+
+            z = result.data
+            fit = result.best_fit
+            bbox = footprint.getBBox()
+            extent = (bbox.getBeginX(), bbox.getEndX(), bbox.getBeginY(), bbox.getEndY())
+            if z.shape[0] == 3:
+                fig = DipolePlotUtils.plt.figure(figsize=(8, 8))
+                for i in range(3):
+                    plt.subplot(3, 3, i*3+1)
+                    DipolePlotUtils.display2dArray(z[i, :], 'Data', True, extent=extent)
+                    plt.subplot(3, 3, i*3+2)
+                    DipolePlotUtils.display2dArray(fit[i, :], 'Model', True, extent=extent)
+                    plt.subplot(3, 3, i*3+3)
+                    DipolePlotUtils.display2dArray(z[i, :] - fit[i, :], 'Residual', True, extent=extent)
+                return fig
+            else:
+                fig = DipolePlotUtils.plt.figure(figsize=(8, 2.5))
+                plt.subplot(1, 3, 1)
+                DipolePlotUtils.display2dArray(z, 'Data', True, extent=extent)
+                plt.subplot(1, 3, 2)
+                DipolePlotUtils.display2dArray(fit, 'Model', True, extent=extent)
+                plt.subplot(1, 3, 3)
+                DipolePlotUtils.display2dArray(z - fit, 'Residual', True, extent=extent)
+                return fig
+        except Exception as err:
+            print('Uh oh! need matplotlib to use these funcs', err)
+            pass
+
+
+######### UTILITIES FUNCTIONS -- TBD WHERE THEY ULTIMATELY END UP ####
```

Comment:
It seems to be the case that this class already has plotting stuff and catalog handling stuff and nothing else. Please split it up accordingly. Is there a reason why the plotting routines can't be in a separate file? Why can't the plot routine from the class above also be in the plotting class?